azurerm/resource_arm_application_gateway_test.go

```diff
@@ -741,6 +741,31 @@ func TestAccAzureRMApplicationGateway_webApplicationFirewall_exclusions(t *testi
 	})
 }
 
+func TestAccAzureRMApplicationGateway_cookieAffinity(t *testing.T) {
+	resourceName := "azurerm_application_gateway.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationGateway_cookieAffinity(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationGatewayExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "backend_http_settings.0.affinity_cookie_name", "testCookieName"),
+				),
```

Comment:
could we also run a step to ensure this value can be removed?

Reply:
Added! 
```
--- PASS: TestAccAzureRMApplicationGateway_cookieAffinity (1239.32s)
```